jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -209,6 +213,45 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Fetches the model name from the modelName ref and then creates a model
+      then switches to it.
+      @method createAndSwitch
+      @param {Object} e The event handler from a form submission.
+    */
+    createAndSwitch: function(e) {
```

Comment:
I feel like this might be better off as a util so that it can be used by other components, however, if this is going to all be removed shortly then we could leave it here. As a util it would also mean you don't have to pass as many props to this component.